stock_location_tank/models/mrp_production.py

```diff
@@ -0,0 +1,16 @@
+# Copyright NuoBiT Solutions - Frank Cespedes <fcespedes@nuobit.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    picking_id = fields.Many2one(comodel_name="stock.picking")
+
+    @api.constrains("product_id", "move_raw_ids", "location_dest_id")
+    def _check_production_lines(self):
+        for production in self:
+            if not production.picking_type_id.tank_filling:
+                super(MrpProduction, self)._check_production_lines()
```

Comment:
Replace `production` for `rec` and call super with `rec` instead of `self`
```python
for rec in self:
     if not rec.picking_type_id.tank_filling:
         super(MrpProduction, rec)._check_production_lines()
```